### ISSUE TITLE
Other dependency reduction in ncdump testing

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#690]
Instead of entering each instance as an issue, here are some of the remaining dependencies that can now be removed in ncdump testing, cleaning up the autotools and cmake build and speeding up the build as well.

* tst_formatx4.sh and tst_formatx3.sh now use different test output names (since the last merge). I believe they are no longer needed in any dependency list.
* run_ncgen_tests.sh and run_ncgen_nc4_tests.sh use the same output file names. Changing them will remove dependencies.
* tst_ncgen_cycle.sh and tst_ncgen_diff.sh are neat scripts that are reused in several tests. They create .nc and .cdl files from a set of test files. Slight changes so that they use unique file names based on the process id ($$) will allow them to run in parallel.
* tst_outputs.sh uses the outputs from the very short run_ctest.sh. These two scripts can combined, removing a test script and a dependency.

In the end, the only dependencies that really make sense are that tst_nccopy.sh and tst_nccopy4.sh run after a bunch of the other tests. These tests use nccopy to copy around a list of files, and those files are produced by other tests.

Otherwise, all dependencies can be eliminated.
